### PR TITLE
Adds an xenobiology access shield wall generator to all maps' xenobiology

### LINKED ISSUE #26087: Scientists don't have the required access to enable the shield wall generators in the xenobiology lab
Server Revision: 8d1ee2267ad7231824b725baffd96daa643cdfc2 

The shield wall generators in xenobiology can't be enabled by the scientist who works there.

### PR DESCRIPTION
**Pull request by BeeSting12**
This adds a new type of shield wall generator, the science one. Looks no different to players, however, it has xenobiology access instead of teleporter so that scientists can use it. 
:cl: BeeSting12
add: Mappers, a science shield wall generator is now available to use in your map's xenobiology.
fix: The shield wall generators in xenobiology can now be locked and unlocked by scientists.
/:cl:

Why: Fixes #26087

### CODE REVIEW

File: _maps/map_files/Cerestation/cerestation.dmm [hunk 1]
```diff
@@ -77224,13 +77224,11 @@
 	},
 /area/toxins/misc_lab)
 "cFm" = (
```
Comment (Supermichael777) [inline _maps/map_files/Cerestation/cerestation.dmm]: hanging bracket

**Comment by optimumtact** [comment]
I haven't seen any proof that they were meant to be accessible for anyone but the RnD, if you can provide of that then this can be considered a fix, but until then this is a rebalance/feature and will be closed.

I will give you 24 hours to find that proof.

**Comment by BeeSting12** [comment]
@optimumtact 
It's common sense that machines in xenobiology should be able to be used by the people working in xeno. If you don't think this PR should be allowed during the freeze, then close it and close the issue so that it doesn't contribute towards the issue count.

**Comment by BeeSting12** [comment]
tldr I have no proof.

**Comment by BeeSting12** [comment]
reopen this if you think that it is a change that should be accepted while the freeze is going

**Comment by KorPhaeron** [comment]
These used to be usable by xenobiologists

**Comment by BeeSting12** [comment]
Wait I actually do have evidence. Metastation's xenobiology currently has the access on it editted so that it should be useable but it's not because he messed up so it's still tele access. Ha!

**Comment by optimumtact** [comment]
Good enough

**Comment by BeeSting12** [comment]
Sweet I'll fix the travis thing tomorrow when I can actually think straight

File: code/game/machinery/shieldgen.dm [hunk 2]
```diff
@@ -247,6 +247,19 @@
...
 	var/storedpower = 0
 
+/obj/machinery/shieldwallgen/science		//Use for xenobiology on maps where you have shield wall generators in xeno.
+	name = "shield wall generator"
+	desc = "A shield generator."
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "Shield_Gen"
+	anchored = 0
+	density = 1
+	req_access = list(access_xenobiology)
```
Comment (Jalleo) [inline code/game/machinery/shieldgen.dm]: wont it only need this as everything else is inherited by the parent?

File: code/game/machinery/shieldgen.dm [hunk 3]
```diff
@@ -247,6 +247,19 @@
 	var/obj/structure/cable/attached		// the attached cable
 	var/storedpower = 0
 
+/obj/machinery/shieldwallgen/science		//Use for xenobiology on maps where you have shield wall generators in xeno.
+	name = "shield wall generator"
```
Comment (Jalleo) [inline code/game/machinery/shieldgen.dm]: Perhaps rename it to specify its for xenobiology

File: code/game/machinery/shieldgen.dm (hunk 2 above)
Comment (BeeSting12) [inline code/game/machinery/shieldgen.dm]: I could delete that, I didn't know how it worked so I thought I had to copy everything in. First time I did it it didn't compile because I pasted the variables too. (still doesn't compile, no clue why but I'll fix when I get home)

File: code/game/machinery/shieldgen.dm (hunk 3 above)
Comment (BeeSting12) [inline code/game/machinery/shieldgen.dm]: I thought about that when I was about to go to bed- I'll rename it to xenobiology shield wall generator when I get home.

**Comment by BeeSting12** [comment]
this web editor is great

**Comment by MMMiracles** [comment]
If you could remove the Cerestation edit, that'd be great. I already have a PR up doing edits on the map and it'd be easier to add this into what I'm doing rather than deal with map conflicts.

**Comment by BeeSting12** [comment]
ok, will do. you'll have to wait till this ones merged to add the xenobiology shield gens though

**Comment by BeeSting12** [comment]
Cerestation is back to how it was before. If I didn't do it right just let me know and I'll redo the whole thing which won't be that hard anyways because fuck me I can't figure out how to revert one file and I'm not even sure if I did it right

**Comment by BeeSting12** [comment]
I'm closing this and making a new one, this time without the cerestation changes and metastation map conflicts. Because I'm pretty sure I screwed up setting the cerestation file back to before the change anyway
